README.md

```diff
@@ -120,6 +123,10 @@ With the default config, APIs will be served over `https` (a self signed certifi
 
 Sample url: `https://localhost:3001/v1/getinfo/`
 
+Providing a `DOMAIN` to the c-lightning-REST configuration will add the domain as a `subjectAltName` to the openssl certificate, permitting successful certificate validation by users and applications, e.g. Zeus, when connecting to the server at via that domain.
+
+If you are *upgrading* a server which is already configured, you will need to first (backup and) delete your `./certs` folder so that new certificates with the new domain will be automatically generated.
```

Comment:
Yes I agree. I did not see a way to do this without breaking existing installations but also agree that this is not made clear in the documentation...

1) Do you see a way for this to be done without breaking existing installs? Would preserving only the macaroon work (I don't know if key info is contained within...

2) if it cannot be done without breaking existing setups, would you like me to add this fact more clearly in the docs, or is this something that would prevent you adding it?

3) I would like to suggest some kind of auto backup and upgrade tool, but I don't think it's worth it for something that already requires so much manual configuration.

Reply:
@willcl-ark Is it possible to add logic to parse the `certs` folder for existing certificate and inspect if the `subjectAltname` property is set. And delete and regenerate the certificate if it doesn't have that property?

This is just the first approach, which comes to mind. I am unsure how easy or difficult it can be.

If you think, this is a reasonable approach. I can also help build this logic.

Reply:
@willcl-ark I tested with your branch, I think we can merge this change, as long as there is no breaking impact on existing installs (which would be the case, since a cert will not be generated if one exists)
Regarding the documentation, we should just indicate that if the user needs to update the cert, only the `certificate.pem` and `key.pem` files should be deleted from the `certs` folder.

Deleting `access.macaroon` will have a huge impact, so that should be avoided.

I will merge this PR into the release branch for the next release (after you've updated the documentation). Thanks.

Reply:
@saubyk OK. I think I've addressed that now in a way which should be adequate for users.

If you don't agree, please provide me with the wording that you'd prefer so that I can put straight in :)

Cheers.

Reply:
Looks good to me.